sdk/src/logs/batch_log_record_processor.cc

```diff
@@ -208,6 +208,8 @@ void BatchLogRecordProcessor::Export()
       break;
     }
 
+    // Reserve space for the number of records
+    records_arr.reserve(num_records_to_export); 
```

Comment:
```suggestion
    records_arr.reserve(num_records_to_export);
```

trailing whitespace.